src/main/java/com/google/devtools/build/lib/bazel/bzlmod/IndexRegistry.java

```diff
@@ -115,7 +117,14 @@ private String constructUrl(String base, String... segments) {
       return Optional.empty();
     }
     String jsonString = new String(bytes.get(), UTF_8);
-    return Optional.of(gson.fromJson(jsonString, klass));
+    if (jsonString.strip().equals("")) {
```

Comment:
why is the empty string treated specially?

also prefer using `String#isEmpty` over `equals("")`

Reply:
Sure, will come with another commit.

The empty string is treated specially because  the `bazel_registry.json` can be empty without being explicitly declared as an empty json object `{}`, this should not throw an exception as it is simply ignored at `getRepoSpec`

```
    if (bazelRegistryJson.isPresent() && bazelRegistryJson.get().mirrors != null) {
      for (String mirror : bazelRegistryJson.get().mirrors) {
        try {
          new URL(mirror);
        } catch (MalformedURLException e) {
          throw new IOException("Malformed mirror URL", e);
        }

        urls.add(constructUrl(mirror, sourceUrl.getAuthority(), sourceUrl.getFile()));
      }
    }
```

Reply:
I'd still rather the file always be valid JSON, but this is such a minor point that I won't push it. Thanks for the contribution!